docs/sdk/client-side-sdks/nextjs/nextjs-gettingstarted.md

```diff
@@ -0,0 +1,268 @@
+'use client'
+import { useVariableValue } from '@devcycle/nextjs-sdk'
+import * as React from 'react'
+
+export const MyClientComponent = function () {
+    const myVariable = useVariableValue('myVariable', false)
+    return (
+        <>
+            <b>Client Variable</b>
+            <span>
+                {JSON.stringify(myVariable)}
+            </span>
+        </>
+    )
+}
+```
+
+### Tracking an event (client component)
+
+```typescript jsx
+'use client'
+import * as React from 'react'
+import { useTrack } from '@devcycle/nextjs-sdk'
+
+export default MyComponent = function () {
+    const trackEvent = useTrack()
+    return (
+        <button onClick={() => trackEvent('myEvent')}>
+            Track Event
+        </button>
+    )
+}
+```
+
+### Tracking an event (server component)
+Currently, tracking events in server components is not supported. Please trigger any event tracking
+from client components.
+
+## Advanced
+### Non-Blocking Initialization
+If you wish to render your page without waiting for the DevCycle configuration to be retrieved, you can use the
+`enableStreaming` option. Doing so enables the following behaviour:
+- the DevCycleServersideProvider will not block rendering of the rest of the server component tree
+- any calls to `getVariableValue` in server components or `useVariableValue` in client components
+  will still block on the config being retrieved. To unblock rendering on these calls,
+  use a `Suspense` boundary to send a fallback while the config is being retrieved. The component will then stream to
+  the client once the config is retrieved.
+
+Note: The DevCycle initialization process is normally very fast (less than 50ms, less than 1ms when cached).
+Only use this option if your application is very performance sensitive.
+
+## Usage (Pages Router)
+### Wrap your app in the DevCycle Higher-Order Component
+In your `_app.tsx` file, wrap the App component in the DevCycle HOC:
+```typescript jsx
+// _app.tsx
+import React from 'react'
+import type { AppProps } from 'next/app'
+import { appWithDevCycle } from '@devcycle/nextjs-sdk/pages'
+
+function MyApp({ Component, pageProps }: AppProps) {
+    return <Component {...pageProps} />
+}
+
+export default appWithDevCycle(MyApp)
+```
+
+In each page in your App where you are using DevCycle, hook up the server-side helper to retrieve
+configuration on the server and allow for server-side rendering using the same user data as the client:
+```typescript jsx
+import { GetServerSideProps } from 'next'
+
+export const getServerSideProps: GetServerSideProps = async (context) => {
+    // get the user identity serverside. Replace with your own function for determining your user's identity
+    const user = {
+        user_id: 'server-user',
+    }
+    return {
+        props: {
+            ...(await getServerSideDevCycle(
+                process.env.NEXT_PUBLIC_DEVCYCLE_CLIENT_SDK_KEY || '',
+                user,
+                context,
+            )),
```

Comment:
This might read a bit clearer: 

```suggestion
            ...(
              await getServerSideDevCycle(
                  process.env.NEXT_PUBLIC_DEVCYCLE_CLIENT_SDK_KEY || '',
                  user,
                  context,
              )
            ),
```

Reply:
this is how prettier formats it